agent/internal/client/mock.go

```diff
@@ -528,5 +533,8 @@ func (c *Mock) GetPullRequestInfo(ctx context.Context, taskData TaskData, prNum
 }
 
 func (c *Mock) CreateInstallationToken(ctx context.Context, td TaskData, owner, repo string) (string, error) {
-	return "token", nil
+	if owner != "" && repo != "" && repo != NotValidRepo {
```

Comment:
Rather than having the mock itself hard-code the result and having to use the hard-coded constant in the test, mocks usually do the opposite - the test that depends on the mock pre-defines a value for the mock to echoed back to it. The dependency injection lets the caller define what the mock is supposed to do, rather than have to write the test to conform to what the mock does, which gives more flexibility in your tests than if the mock decides the output.

[This](https://github.com/evergreen-ci/evergreen/blob/ff46048d63feb595caedb68fd4d9fc85ab73a96b/agent/internal/client/mock.go#L384-L390) is one example of that pattern being used. The test can set `(MockCommunicator).GetTaskPatchResponse` and whatever testing code depends on the patch will use the test's desired patch.

Reply:
updated